### PR TITLE
changes to config for noetic port

### PR DESCRIPTION
**Pull request by pmusau17**
# PR Details

Changes to the docker compose files to test out the noetic port, this also includes a set of scripts to run the carma platform without breaking any of the main carma bash file. Primarily this pull request is to test if the noetic build is reproducible.

## Description

I had to change the docker compose file. Included a set of bash scripts.
## Related Issue

There are commented out sections of the original build files. I still need to work on fixing those/ need guidance as how they want this done.

## Motivation and Context

Building on Noetic is an intermediate step towards ROS2 port.

## How Has This Been Tested?

Tested on Ubuntu 20.04 laptop and virtual machine.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: development/build-image.sh [hunk 1]
```diff
@@ -53,24 +53,28 @@ fi
...
-if [[ $TAG = "develop-$CONFIG_NAME" ]]; then
-    git checkout -- docker-compose.yml
-    sed -i "s|usdotfhwastoldev/|$USERNAME/|g; s|usdotfhwastolcandidate/|$USERNAME/|g; s|usdotfhwastol/|$USERNAME/|g; s|:[0-9]*\.[0-9]*\.[0-9]*|:develop|g; s|:CARMASystem_[0-9]*\.[0-9]*\.[0-9]*|:develop|g;" \
-        docker-compose.yml
-    sed -i "s|usdotfhwastoldev/|$USERNAME/|g; s|usdotfhwastolcandidate/|$USERNAME/|g; s|usdotfhwastol/|$USERNAME/|g; s|:[0-9]*\.[0-9]*\.[0-9]*|:develop|g; s|:CARMASystem_[0-9]*\.[0-9]*\.[0-9]*|:develop|g;" \
-        docker-compose-background.yml
-else
-    git checkout -- docker-compose.yml docker-compose-background.yml
-fi
+# if [[ $TAG = "develop-$CONFIG_NAME" ]]; then
```
Comment (mjeronimo) [inline development/build-image.sh:56]: Remove any dead blocks.

File: development/build-image.sh [hunk 2]
```diff
@@ -53,24 +53,28 @@ fi
...
     --build-arg VCS_REF=`git rev-parse --short HEAD` \
     --build-arg CONFIG_NAME="carma-config:$CONFIG_NAME" \
     --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` .
 
+docker tag $USERNAME/$IMAGE:$TAG $USERNAME/$IMAGE:latest
+TAGS+=("$USERNAME/$IMAGE:latest")
+
 # restore docker-compose.yml in case edited by if statement
-git checkout -- docker-compose.yml
+# git checkout -- docker-compose.yml
```
Comment (mjeronimo) [inline development/build-image.sh:77]: Remove if not needed.

File: development/build-image.sh [hunk 3]
```diff
@@ -53,24 +53,28 @@ fi
...
+# fi
+
 
 docker build --no-cache -t $USERNAME/$IMAGE:$TAG \
     --build-arg VERSION="$TAG" \
     --build-arg VCS_REF=`git rev-parse --short HEAD` \
     --build-arg CONFIG_NAME="carma-config:$CONFIG_NAME" \
     --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` .
 
+docker tag $USERNAME/$IMAGE:$TAG $USERNAME/$IMAGE:latest
```
Comment (mjeronimo) [inline development/build-image.sh:73]: Is this correct? Should it be 'develop' ?

File: development/docker-compose.yml [hunk 4]
```diff
@@ -17,7 +17,7 @@ version: '2'
 
 services:
   roscore:
-    image: usdotfhwastoldev/carma-base:develop
+    image: usdotfhwastol/carma-base:latest
```
Comment (mjeronimo) [inline development/docker-compose.yml:20]: I think we'll want to stay with 'develop'

File: development/set_local_config.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    carma__stop
+    echo "Deleting old CARMA config..."
+    docker rm carma-config
+fi
+echo "Setting $IMAGE_NAME as current CARMA configuration instance..."
+docker run --name carma-config "$IMAGE_NAME"
+
+echo "Making sure that $IMAGE_NAME has the correct docker-compose-file..."
+docker run --rm --volumes-from carma-config:ro --entrypoint sh busybox:latest -c \
+        'cat /opt/carma/vehicle/config/docker-compose.yml'
```
Comment (mjeronimo) [inline development/set_local_config.sh:16]: Looks like it wants a newline

File: development/start_up_noetic.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# start up background processes 
+docker-compose -f docker-compose-background.yml -p carma-background up -d
+
+# start up container
+docker-compose -f docker-compose.yml -p carma up
```
Comment (mjeronimo) [inline development/start_up_noetic.sh:7]: newline

**Review by mjeronimo — CHANGES_REQUESTED** [review]

**Comment by pmusau17** [comment]
Changes no longer needed

**Pull request closed without merging**